shards/shards_test.go

```diff
@@ -958,6 +959,175 @@ func TestAtomCountScore(t *testing.T) {
 	}
 }
 
+func TestSearchFileTombstones(t *testing.T) {
```

Comment:
I'd also appreciate feedback on the tests. One thing that strikes me as odd is that asserting on the number of line matches seems brittle / maybe asserting too much about internals.

Reply:
two comments:

- you can have a single corpus of documents / repository. Then each test case only alternates what you care about `FileTombstones`. Smaller test descriptions lead to easier grokking of what the test is doing.
- Yeah line match counts doesn't give me confidence. I think maybe you should assert on the set of filenames returned.